projects/core/src/user/store/effects/user-details.effect.ts

```diff
@@ -26,6 +27,16 @@ export class UserDetailsEffects {
     })
   );
 
+  @Effect()
```

Comment:
I'm not sure I understand this part, why do we need this?

Reply:
There are no user information sent back with the auth token, we need to query the user details if only to get the customerId we need to use in all the subsequent calls to occ ( or the uid, even though we used to artificially store it with the user token state)

Basically, what we do here is that we say that loading the user details is part of the authentication data loaded upon authentication.  After authentication, the app is guaranteed to have user details in the state.  Components can rely on the fact that user details for the authenticated user is available when a user is authenticated and can base some logic around this fact.

This also sets the stage for ASM support where we initially load the customer agent support agent’s user details, but afterwards will switch to other customer details without changing the token when an agent impersonates a user.  So we need that broader authentication footprint.